File: 2022_c#_basics/12-while-loop-exercise/06-cake.cs
using System;

namespace _06_Cake
{
    internal class Program
    {
        static void Main(string[] args)
        {
            int l = int.Parse(Console.ReadLine());
            int w = int.Parse(Console.ReadLine());
            double cake = l * w;
            double pieces = 0.0;

            while (cake > 0)
            {
                string input = Console.ReadLine();

                if (input == "STOP")
                {
                    break;
                }

                int takenPieces = int.Parse(input);
                cake -= takenPieces;
            }

            if (cake >= 0)
            {
                Console.WriteLine($"{cake} pieces are left.");
            }
            else
            {
                Console.WriteLine($"No more cake left! You need {Math.Abs(cake)} pieces more.");
            }
        }
    }
}

File: 2024_java_basics/12-while-loops-lab/tasks/06-cake.java
import java.util.Scanner;

public class Cake {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int length = Integer.parseInt(scanner.nextLine());
        int width = Integer.parseInt(scanner.nextLine());
        double cake = length * width;
        double pieces = 0.0;

        while (cake > 0) {
            String input = scanner.nextLine();

            if (input.equals("STOP")) {
                break;
            }

            int takenPieces = Integer.parseInt(input);
            cake -= takenPieces;
        }

        if (cake >= 0) {
            System.out.printf("%.0f pieces are left.%n", cake);
        } else {
            System.out.printf("No more cake left! You need %.0f pieces more.%n", Math.abs(cake));
        }

        scanner.close();
    }
}
